services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/system_monitor/_disk_usage.py

```diff
@@ -17,41 +21,177 @@
 from ..mounted_fs import MountedVolumes
 from ..notifications import publish_disk_usage
 
+_NODE_FILE_SYSTEM_PATH: Final[Path] = Path("/")
+
+
 _logger = logging.getLogger(__name__)
 
 
+class MountPathCategory(StrAutoEnum):
+    HOST = auto()
+    STATES_VOLUMES = auto()
+    INPUTS_VOLUMES = auto()
+    OUTPUTS_VOLUMES = auto()
+
+
+_SUPPORTED_ITEMS: Final[set[str]] = {
+    MountPathCategory.HOST,
+    MountPathCategory.STATES_VOLUMES,
+}
+
+
 async def get_usage(path: Path) -> DiskUsage:
     usage = await asyncio.get_event_loop().run_in_executor(
         None, psutil.disk_usage, f"{path}"
     )
     return DiskUsage.from_ps_util_disk_usage(usage)
 
 
+def get_relative_path(path: Path, dy_volumes_mount_dir: Path) -> Path:
+    try:
+        return path.relative_to(dy_volumes_mount_dir)
+    except ValueError:
+        return path
+
+
+def _get_normalized_folder_name(path: Path) -> str:
+    return f"{path}".replace("/", "_")
+
+
+def _have_common_entries(a: set[str], b: set[str]) -> bool:
+    return len(a & b) > 0
+
+
 @dataclass
 class DiskUsageMonitor:
     app: FastAPI
     user_id: UserID
     node_id: NodeID
     interval: timedelta
-    monitored_paths: list[Path]
+    monitored_paths: dict[MountPathCategory, set[Path]]
+
+    dy_volumes_mount_dir: Path
     _monitor_task: asyncio.Task | None = None
-    _last_usage: dict[Path, DiskUsage] = field(default_factory=dict)
+    _incoming_overwrite_usage: dict[str, DiskUsage] = field(default_factory=dict)
+    _last_usage: dict[str, DiskUsage] = field(default_factory=dict)
+
+    @cached_property
+    def _flat_monitored_paths(self) -> set[Path]:
+        return set.union(*self.monitored_paths.values())
+
+    @cached_property
+    def _normalized_monitored_paths(self) -> dict[MountPathCategory, set[str]]:
```

Comment:
is this called normalization? you are replacing slashes with underscores?

Reply:
Correct, this one way (Path -> str) transformation is used to uniquely indetify the folder in the `/dy-volumes` path. Which is also what the eft-guardian does.

Reply:
I've added a bit of docstring to help with this